tripy/nvtripy/trace/ops/dequantize.py

```diff
@@ -33,65 +34,13 @@ class Dequantize(BaseTraceOp):
     def infer_dtypes(self):
         self.outputs[0].dtype = self.dtype
 
-    def to_flat_ir(self, inputs, outputs):
-        from nvtripy.common.datatype import int32
-        from nvtripy.flat_ir.ops import ConcatenateOp, ConvertOp, DivideOp, DynamicBroadcastOp, DynamicReshapeOp, MulOp
-        from nvtripy.flat_ir.tensor import FlatIRTensor
-
-        # Represent quantize as convert(input, dtype) * scale
-        converted_tensor = FlatIRTensor.build(
-            shape=inputs[0].shape,
-            rank=inputs[0].rank,
-            dtype=self.dtype,
-            device=inputs[0].device,
-            reason_details=["Convert the input tensor to dequantized dtype."],
-        )
-        ConvertOp.build([inputs[0]], [converted_tensor])
-
-        broadcast_scale = FlatIRTensor.build(
-            shape=inputs[0].shape,  # broadcast to input's shape
-            rank=inputs[0].rank,
-            dtype=inputs[1].dtype,  # original scale's dtype
-            device=inputs[1].device,
-            reason_details=["Broadcast the scale to the input's shape in dequant operation."],
-        )
-        if inputs[1].rank == 0 or inputs[1].rank == 1:
-            shape_of_input = op_utils.get_shape_of_tensor(inputs[0])
-            broadcast_dim = [self.dim] if self.dim is not None else []
-            DynamicBroadcastOp.build([inputs[1], shape_of_input], [broadcast_scale], broadcast_dim=broadcast_dim)
-        else:
-            # block-wise quant, input: [block_size * A, B], scale: [A, B]
-            # Broadcast(scale) -> [block_size, A, B]
-            # Reshape(scale) -> [block_size * A, B]
-            # Mul(input, scale)
-            num_blocks = FlatIRTensor.build(
-                shape=(1,),
-                rank=1,
-                dtype=int32,
-                device=inputs[0].device,
-                reason_details=["Compute the number of blocks in block-wise dequantization"],
-            )
-            blocked_shape = FlatIRTensor.build(
-                shape=(3,),
-                rank=1,
-                dtype=int32,
-                device=inputs[0].device,
-                reason_details=["Compute shape with an extra blocked_size dimension."],
+    def to_mlir(self, inputs, outputs):
+        assert len(inputs) == 2, "Dequantize operation should have exactly 2 inputs!"
+        return [
+            tensorrt.dequantize(
+                result=self.outputs[0].to_mlir(),
```

Comment:
You can just access `outputs` instead of converting again:
```suggestion
                result=outputs[0],
```